Sources/SwiftDiagnostics/Diagnostic.swift

```diff
@@ -40,17 +40,17 @@ public struct Diagnostic: CustomDebugStringConvertible, Sendable {
   /// If `highlights` is `nil` then `node` will be highlighted. This is a
   /// reasonable default for almost all diagnostics.
   public init(
-    node: some SyntaxProtocol,
+    node: (some SyntaxProtocol)?,
     position: AbsolutePosition? = nil,
     message: DiagnosticMessage,
     highlights: [Syntax]? = nil,
     notes: [Note] = [],
     fixIts: [FixIt] = []
   ) {
-    self.node = Syntax(node)
-    self.position = position ?? node.positionAfterSkippingLeadingTrivia
+    self.node = node.map { Syntax($0) }
```

Comment:
I think we have a `Syntax` initializer that just proposes `nil`, so it should be possible to leave this as-is.